### ISSUE TITLE
Cannot use capture tag inside a liquid tag

### ISSUE THREAD

**Issue by pdehaan** [harttle/liquidjs#552]
Using 9.42.1 this code works:

```liquid
{% assign favorite_food = "pizza" %}
{% assign age = 35 %}
{% capture about_me %}
  I am {{ age }} and my favorite food is {{ favorite_food }}.
{% endcapture %}

{% liquid
  # render 'header.liquid' for names as name
  echo about_me
%}
```

But if I try using a `capture` tag within a `liquid` tag, I get errors:

```liquid
{% liquid
  assign favorite_food2 = "pizza"
  assign age2 = 35
  capture about_me2
    I am nnn and my favorite food is nnn.
  endcapture

  echo about_me2
%}
```

### ERROR (via Eleventy 1.0.2)

```
> eleventy --quiet

[11ty] Problem writing Eleventy templates: (more in DEBUG output)
[11ty] 1. Having trouble rendering liquid template ./src/index.liquid (via TemplateContentRenderError)
[11ty] 2. Having trouble compiling template ./src/index.liquid (via TemplateContentCompileError)
[11ty] 3. tag "I" not found, line:4, col:1, line:3, col:1, file:./src/index.liquid, line:32, col:1 (via ParseError)
[11ty] 4. tag "I" not found, line:4, col:1, line:3, col:1 (via ParseError)
[11ty] 5. tag "I" not found, line:4, col:1 (via ParseError)
[11ty] 6. tag "I" not found (via AssertionError)
[11ty] 
[11ty] Original error stack trace: AssertionError: tag "I" not found
[11ty]     at new AssertionError (/private/tmp/11ty-liquid-tag/node_modules/liquidjs/dist/liquid.node.cjs.js:1198:28)
[11ty]     at assert (/private/tmp/11ty-liquid-tag/node_modules/liquidjs/dist/liquid.node.cjs.js:1233:15)
[11ty]     at TagMap.get (/private/tmp/11ty-liquid-tag/node_modules/liquidjs/dist/liquid.node.cjs.js:4617:9)
[11ty]     at new Tag (/private/tmp/11ty-liquid-tag/node_modules/liquidjs/dist/liquid.node.cjs.js:4439:32)
[11ty]     at Parser.parseToken (/private/tmp/11ty-liquid-tag/node_modules/liquidjs/dist/liquid.node.cjs.js:4535:24)
[11ty]     at ParseStream.parseToken (/private/tmp/11ty-liquid-tag/node_modules/liquidjs/dist/liquid.node.cjs.js:4548:80)
[11ty]     at ParseStream.start (/private/tmp/11ty-liquid-tag/node_modules/liquidjs/dist/liquid.node.cjs.js:2628:33)
[11ty]     at Object.parse (/private/tmp/11ty-liquid-tag/node_modules/liquidjs/dist/liquid.node.cjs.js:3019:16)
[11ty]     at new Tag (/private/tmp/11ty-liquid-tag/node_modules/liquidjs/dist/liquid.node.cjs.js:4443:24)
[11ty]     at Parser.parseToken (/private/tmp/11ty-liquid-tag/node_modules/liquidjs/dist/liquid.node.cjs.js:4535:24)
[11ty] Wrote 0 files in 0.03 seconds (v1.0.2)
```

```js
const { Liquid } = require('liquidjs')
const engine = new Liquid()
const tpl = engine.parse(`
{% liquid
  assign favorite_food2 = "pizza"
  assign age2 = 35
  capture about_me2
    I am nnn and my favorite food is nnn.
  endcapture

  echo about_me2
%}
`)
engine.render(tpl).then(console.log)
// originalError: Error: tag "I" not found, line:4, col:1, line:3, col:1
```

**Comment by harttle** [comment]
I think `capture` is not well supported in `liquid` tag. The ruby version throws a similar error on your template:

```
Liquid syntax error (line 7): Unknown tag 'I' (Liquid::SyntaxError)
```

I can follow this topic and hopefully the final solution can be consistent: https://github.com/Shopify/liquid/issues/1373 . Will leave this issue open for discussion until then.

**Comment by timdmackey** [comment]
This code sample is invalid, and would throw an error in vanilla liquid as well:
> ```liquid
> {% liquid
>   assign favorite_food2 = "pizza"
>   assign age2 = 35
>   capture about_me2
>     I am nnn and my favorite food is nnn.
>   endcapture
> 
>   echo about_me2
> %}
> ```
The line `I am nn...` needs to be preceded by an `echo` tag and wrapped in quotes to make this valid. In fact, any line inside a `{% liquid %} tag with non-whitespace characters must be preceded by a tag, and the tag must be closed at the end of the line.

I haven't used liquidjs to know if this code would work here, but it would work correctly in Shopify liquid:

```liquid
{% liquid
  assign favorite_food2 = "pizza"
  assign age2 = 35
  capture about_me2
    echo "I am nnn and my favorite food is nnn."
  endcapture

  echo about_me2
%}
```

**Comment by harttle** [comment]
Thank you @timdmackey  for clarification. I can confirm the above snippet also works in the latest LiquidJS. See this [live demo](https://liquidjs.com/playground.html#eyUgbGlxdWlkDQogIGFzc2lnbiBmYXZvcml0ZV9mb29kMiA9ICJwaXp6YSINCiAgYXNzaWduIGFnZTIgPSAzNQ0KICBjYXB0dXJlIGFib3V0X21lMg0KICAgIGVjaG8gIkkgYW0gbm5uIGFuZCBteSBmYXZvcml0ZSBmb29kIGlzIG5ubi4iDQogIGVuZGNhcHR1cmUNCg0KICBlY2hvIGFib3V0X21lMg0KJX0=,e30=).

Closing since it is by design, at least on par with Shopify liquid. But I noticed replacing the `nnn` with `{{age}}` won't work in both LiquidJS and Shopify liquid. Feel free to open another issue if there's actual a a way for this to work in Shopify liquid.
